### ISSUE TITLE
renaming a notebook folder throws a lot of errors

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#2598]
It looks like the rename succeeds but then it throws an error before it is able to update the folder tree.

Clicking on each notebook pulls in the new title and the notebook moves in the specified folder.

The error is 

```
TypeError: editor.update_notebook_from_gist is not a function
at https://rcloud-5.research.att.com/js/rcloud_bundle.js:10856:36
From previous event:
at https://rcloud-5.research.att.com/js/rcloud_bundle.js:10855:26
at Object.for_each_notebook (https://rcloud-5.research.att.com/editor_tab.js:239:17)
at https://rcloud-5.research.att.com/editor_tab.js:234:26
at Array.forEach (<anonymous>)
at Object.for_each_notebook (https://rcloud-5.research.att.com/editor_tab.js:233:31)
at https://rcloud-5.research.att.com/js/rcloud_bundle.js:10850:20
at execute_if_valid_else_ignore (https://rcloud-5.research.att.com/js/rcloud_bundle.js:1324:29)
at HTMLSpanElement.keydown.editable (https://rcloud-5.research.att.com/js/rcloud_bundle.js:1336:32)
at HTMLSpanElement.dispatch (https://rcloud-5.research.att.com/lib/js/jquery-2.1.1.js:4409:9)
at HTMLSpanElement.elemData.handle (https://rcloud-5.research.att.com/lib/js/jquery-2.1.1.js:4095:28)
```

Looks like it's a consequence of the big refactor that led to tree sorting and filtering.

**Comment by gordonwoodhull** [comment]
To be clear, since this feature is not obvious: you can click on a notebook folder when it is open, and rename it. This is shorthand for renaming each of the notebooks inside of it.

**Comment by JABedford** [comment]
Having trouble with this apparently being a very sporadic issue @gordonwoodhull. Do you experience the same or is this consistent for you?

My PR request seemingly moves the error from the interface into the browser console, and once it runs two or three times it will work absolutely fine for me on both Firefox and Chrome, hence me believing it was solved. 

It did spin up #2577 for me a few times on test however.

**Comment by gordonwoodhull** [comment]
The browser console is what you need to look at here - it's not guaranteed that every error will be reflected in the Session panel. Normally there should not be any errors in the browser console at all.

I think the underlying renames are working okay but it is not reflected in the UI or metadata. Additionally, the current notebook is correctly and immediately renamed - so that probably explains why you see it working fine sometimes. You have to have multiple notebooks in the folder in order to see the breakage.

An indication that the underlying rename worked: once you click on the other notebooks, they are also moved to the correct folder.

What is going on is that we keep a cache of notebook metadata - mostly names (including folders) and dates in RCS, which is our Redis database. So if an error occurs, you can end up in a situation where the underlying notebook gist (stored on GitHub or our own rcloud-gist-services) has a different name from what is in RCS or the displayed notebook tree.

**Comment by gordonwoodhull** [comment]
Fixed by #2620
